src/app/campaign/flight/flight.container.ts

```diff
@@ -40,23 +54,28 @@ export class FlightContainerComponent implements OnInit {
     );
   }
 
-  ngOnInit() {}
+  ngOnDestroy() {
+    if (this.flightSub) { this.flightSub.unsubscribe(); }
+  }
 
-  setFlightId(id: string) {
-    this.campaignStoreService.campaignFirst$.subscribe(state => {
-      if (state.flights[id]) {
-        this.currentFlightId = id;
-        this.state$.next(state.flights[id]);
-        this.currentInventoryUri$.next(state.flights[id].localFlight.set_inventory_uri);
-      } else {
-        const campaignId = state.remoteCampaign ? state.remoteCampaign.id : 'new';
-        this.router.navigate(['/campaign', campaignId]);
+  setFlightId(id: string, state: CampaignState) {
+    if (state.flights[id]) {
+      if (this.currentFlightId !== id) {
```

Comment:
I'm guessing this logic is needed because of our decision not to store the temporary flight IDs we create for local flights before they're persisted in the `id` field of a `Flight`. I'm not super convinced this was the right decision, as I'm seeing special case logic like this pop up in a couple of places in my work, and also in this PR. Worth keeping an eye on.

Reply:
I haven't run across the temporary ids. I wanted to loadAvailability when the flight id param changes. Also I want to set the Observable for availability rollup according to the flight id param. Initially I added the currentFlightId onto the store state and used that in the availability rollup, but I decided I could just leave the current flight id here in the container component for now.